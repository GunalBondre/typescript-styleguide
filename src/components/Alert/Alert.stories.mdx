import { Canvas, Meta, Story } from '@storybook/addon-docs';
import Alert from './Alert';

<Meta title='Design System/Components/Alert' component={Alert} />

# Alert component

#### Basic Alert component

<Canvas>
	<Story name='Success'>
		<Alert variant='success' message='Success alert' />
	</Story>
</Canvas>

#### Basic Alert warning component

<Canvas>
	<Story name='Warning'>
		<Alert variant='warning' message='warning alert' />
	</Story>
</Canvas>

#### Basic Alert error component

<Canvas>
	<Story name='Error'>
		<Alert variant='error' message='error alert' />
	</Story>
</Canvas>

#### Basic Alert general component

<Canvas>
	<Story name='General'>
		<Alert variant='general' message='general alert' />
	</Story>
</Canvas>

<Canvas>
	<Story name='Action_Button'>
		<Alert
			variant='warning'
			message='general alert'
			action_button
			action_button_text='Click Me'
		/>
	</Story>
</Canvas>
