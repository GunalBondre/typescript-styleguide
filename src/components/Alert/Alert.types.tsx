export interface AlertProps {
	variant: string;
	message: string;
	title?: string;
	custom_radius?: string;
	action_button?: string;
	bgColor?: string;
	textColor?: string;
	customWidth?: string;
	children?: any;
	props?: JSX.Element;
}

interface AlertInterface {
	SUCCESS: string;
	WARNING: string;
	ERROR: string;
	GENERAL: string;
}

export const AlertTypes: AlertInterface = {
	SUCCESS: 'success',
	WARNING: 'warning',
	ERROR: 'error',
	GENERAL: 'general',
};
