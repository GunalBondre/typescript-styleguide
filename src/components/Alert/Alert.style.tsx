import styled from 'styled-components';
import { AlertProps } from './Alert.types';

import { theme } from '../../themes/theme';

export const AlertWrapper = styled.div<AlertProps>`
	display: flex;
	background-color: ${(props): any =>
		props.bgColor ? props.bgColor : theme.Alert[props.variant].bgColor};
	padding: 10px 20px;
	color: ${(props) => theme.Alert[props.variant].textColor};
	border-radius: ${(props) =>
		props.custom_radius ? props.custom_radius : 5}px;
`;

export const MessageIconWrapper = styled.div``;
export const ButtonWrapper = styled.div<AlertProps>``;
