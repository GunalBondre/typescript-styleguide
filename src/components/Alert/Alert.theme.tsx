import { AlertTypes } from './Alert.types';
import { Palette } from '../../themes/palette';

export const AlertTheme = {
	[AlertTypes.SUCCESS]: {
		background: Palette.green200,
	},
};
