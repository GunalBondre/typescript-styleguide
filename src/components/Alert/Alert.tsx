/* eslint-disable react/jsx-no-comment-textnodes */
import React, { FC } from 'react';
import { AlertWrapper, ButtonWrapper, MessageIconWrapper } from './Alert.style';
import { AlertProps } from './Alert.types';

const Alert: FC<AlertProps> = (props): JSX.Element => {
	const {
		variant,
		message,
		title,
		custom_radius,
		bgColor,
		customWidth,
		action_button,
		textColor,
		...restProps
	} = props;
	return (
		<AlertWrapper
			variant={variant}
			message={message}
			title={title}
			customWidth={customWidth}
			bgColor={bgColor}
			custom_radius={custom_radius}
			action_button={action_button}
			textColor={textColor}
			data-testid='alert'
			{...restProps}
		>
			<MessageIconWrapper>{message}</MessageIconWrapper>
		</AlertWrapper>
	);
};

export default Alert;
