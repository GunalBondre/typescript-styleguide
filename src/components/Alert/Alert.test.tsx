import { render, screen } from '@testing-library/react';
import Alert from './Alert';

describe('Alert component', () => {
	it('should render success alert component', () => {
		render(<Alert variant='success' message='success alert' />);

		expect(screen.getByTestId('alert')).toBeDefined();
	});
	it('should render error alert component', () => {
		render(<Alert variant='error' message='error alert' />);

		expect(screen.getByTestId('alert')).toBeDefined();
	});
	it('should render warning alert component', () => {
		render(<Alert variant='warning' message='warning alert' />);

		expect(screen.getByTestId('alert')).toBeDefined();
	});
	it('should render general alert component', () => {
		render(<Alert variant='general' message='general alert' />);

		expect(screen.getByTestId('alert')).toBeDefined();
	});
	it('should render alert component with button and text', () => {
		render(
			<Alert
				variant='general'
				message='general alert'
				action_button
				action_button_text='click Me'
			/>
		);

		expect(screen.getByTestId('alert')).toBeDefined();
	});
});
