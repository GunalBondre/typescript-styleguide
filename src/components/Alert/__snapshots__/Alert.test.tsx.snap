// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert component should render success alert component 1`] = `
<div>
  <div
    class="sc-bczRLJ fXwCea"
    data-testid="alert"
  >
    <div
      class="sc-gsnTZi"
    >
      success alert
    </div>
  </div>
</div>
`;
